flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/io/benchmark/StreamNetworkBenchmarkEnvironment.java

```diff
@@ -272,11 +287,161 @@ private static ShuffleDescriptor createShuffleDescriptor(
 			ResultPartitionID resultPartitionID,
 			ResourceID location,
 			TaskManagerLocation senderLocation,
-			int channel) {
+			int connectionIndex) {
 		final NettyShuffleDescriptorBuilder builder = NettyShuffleDescriptorBuilder.newBuilder()
 			.setId(resultPartitionID)
 			.setProducerInfoFromTaskManagerLocation(senderLocation)
-			.setConnectionIndex(channel);
+			.setConnectionIndex(connectionIndex);
 		return localMode ? builder.setProducerLocation(location).buildLocal() : builder.buildRemote();
 	}
+
+	/**
+	 * A {@link SingleInputGateFactory} which replaces the default {@link RemoteInputChannel} and
+	 * {@link LocalInputChannel} implementation with costume ones.
+	 */
+	private static class TestSingleInputGateFactory extends SingleInputGateFactory {
+
+		private final ResourceID taskExecutorResourceId;
+		private final int partitionRequestInitialBackoff;
+		private final int partitionRequestMaxBackoff;
+		private final ConnectionManager connectionManager;
+		private final ResultPartitionManager partitionManager;
+		private final TaskEventPublisher taskEventPublisher;
+		private final NetworkBufferPool networkBufferPool;
+
+		public TestSingleInputGateFactory(
+				@Nonnull ResourceID taskExecutorResourceId,
```

Comment:
nit: I am a bit torn for `@Nonnull` with arguments. I know we did it in some cases before like `SingleInputGateFactory`. 
In common sense we still consider them as `Nonnull` w/o this annotation. And we only need to emphasis the `Nullable` case to add the annotation.

In another aspect, it seems not very consistent in this class because other constructors do not have it.